### ISSUE TITLE
Add Beautify on Save

### ISSUE THREAD

**Issue by jayrosen1576** [HookyQR/VSCodeBeautify#5]
I know VS Code does some beautification as text it entered into the editor but a beautify on save feature would be great for files that are opened but not changed. Can this be added to this extension or is it more suited as a feature in the main VSCode app? If it is something easy to add then it would be great to have it in your extension.

**Comment by HookyQR** [comment]
Would you be happy with this as a mutable set of workspace setting:

```javascript
//beautify all css, js, html on save/close
"beautify.beautifyOnClose": true 
// or
"beautify.beautifyOnClose":  [
  "directory/**/beautify",
  "!directory/**/not/beautify",
  "!**/*.min.js" ]
// or
"beautify.beautifyOnClose": {
  // all css
  "css": true,
  // matches 'directory/needed/to/beautify' but not 'directory/must/not/beautify'
  // and never beautifies .min.js files
  "js": [
    "directory/**/beautify",
    "!directory/**/not/beautify",
    "!**/*.min.js" ]
  // don't do html (simply leaving this out would do the same)
  "html": false
}
```

Using the [.gitignore pattern matching](http://git-scm.com/docs/gitignore) (as the VS Code workspace search does). Especially the negation syntax would be useful.

I'm not so sure about beautifying on close vs on save. I would expect opening, then closing a file to have no side effect. Having a beautifyOnSave instead would simply require you to hit save, or just enter a space and let the saving (prompted, or auto) do the work.

Thoughts?

**Comment by jayrosen1576** [comment]
I think just a simple beautifyOnSave for the currently focused document would suffice. this is how sublimetext/atom do it and it works well. I would have to be tied to the save function not close however, since I often want to keep the file open after it is formatted. Basically if you can invoke the "editor.action.format" when cmd+s / ctrl+s is pressed that would be excellent.

**Comment by HookyQR** [comment]
Added functionality as requested.

Published as `0.1.0`. Take a look at the readme for user settings.

It won't run if you close a dirty file from the working files list, or if you quit VS Code with a dirty file active. The required trigger function is not called during the save for these processes.

**Comment by jayrosen1576** [comment]
I love it! Thanks for putting this in so quickly! Question: I use a gulp script for minify files into a public directory. The beautifyOnSave should not affect these correct? In other words, this feature is only activated when you save a file (cmd+s) within the editor window right?

**Comment by HookyQR** [comment]
There is a `beautify.onSaveIgnore` setting where you can put your files/folders to be ignored. By default it will ignore all files ending in `.min.*`, `_min.*`, or `-min.*`. Have a look at the readme for details. If you need help setting up a glob path for you setup let me know.

**Comment by HookyQR** [comment]
Implemented in 0.1.0
